### DISCUSSION TITLE
Parse a value from a nested dictionary

### DISCUSSION THREAD

**Discussion by mohammadali95** [pydantic/pydantic#8098] [Question]
Hey so I have a data object that looks like below:
`request = {'name_alias' : 'ali',  'attributes': {'frst_nm': 'Johny'}}`

I need to extract `frst_nm` from my data input to include in the model. Here is my model:
`class SavePersonAtaccamaModel(BaseModel):`
    `name: str = Field(alias='name_alias')`
    `UserName: str = Field(source='attributes.frst_nm')`

However, this doesn't seem to work. Any idea how I can reference the nested dict in my model?  I want the final dict after validation to look like below:
`{name: ali, UserName: Johny}`

**Comment by fdemmer** [comment]
Use an `AliasPath`: https://docs.pydantic.dev/latest/concepts/alias/#aliaspath-and-aliaschoices

```python
class SavePersonAtaccamaModel(BaseModel):
    name: str = Field(alias='name_alias')
    UserName: str = Field(validation_alias=AliasPath('attributes', 'frst_nm'))
```
